[docs] Update `command-line-interface.md` to new guidelines (#5880)
DevExpress/testcafe: docs/articles/documentation/reference/command-line-interface.md

  * [--color](#--color)
  * [--no-color](#--no-color)

> When you execute the `testcafe` command, TestCafe first reads settings from the `.testcaferc.json` [configuration file](configuration-file.md) if this file exists, and then applies the settings from the command line. Command-line settings override conflicting values from the configuration file. TestCafe prints information about every overridden property in the console.

Author: If TestCafe is launched with a CLI parameter, and the very same parameter is present in the config, there is a conflict.
CLI overrides what's in the config in this case.

Does this work:

> When you execute the `testcafe` command, TestCafe first reads settings from the `.testcaferc.json` configuration file if this file exists, and then applies the settings from the command line. Command-line settings override values from the configuration file. TestCafe prints information about every overridden property in the console.

> When you execute the `testcafe` command, TestCafe first reads settings from the `.testcaferc.json` [configuration file](configuration-file.md) if this file exists, and then applies the settings from the command line. Command-line settings override conflicting values from the configuration file. TestCafe prints information about every overridden property in the console.

Reviewer: When you execute the testcafe command, TestCafe first reads settings from the .testcaferc.json configuration file if this file exists, and then applies the settings from the command line. In there are conflicting values, a command-line setting overrides these values in the configuration file. TestCafe prints information about every overridden property in the console.

> When you execute the `testcafe` command, TestCafe first reads settings from the `.testcaferc.json` [configuration file](configuration-file.md) if this file exists, and then applies the settings from the command line. Command-line settings override conflicting values from the configuration file. TestCafe prints information about every overridden property in the console.

Author: ```suggestion
When you execute the `testcafe` command, TestCafe first reads settings from the `.testcaferc.json` [configuration file](configuration-file.md) if this file exists, and then applies the settings from the command line. If there are conflicting values, command-line settings override these values in the configuration file. TestCafe prints information about every overridden property in the console.
```

> If you run tests [concurrently](#-c-n---concurrency-n),
> specify the total number of all browsers' instances after the `remote:` keyword.

> You can use the [--qr-code](#--qr-code) option to display QR-codes that represent the same URLs.

Author: QR codes contain information (URLs in this case). Is that more accurate:

> You can use the `--qr-code` option to display QR-codes that contain the URLs.

> You can use the [--qr-code](#--qr-code) option to display QR-codes that represent the same URLs.

Reviewer: Ok

> You can use the [--qr-code](#--qr-code) option to display QR-codes that represent the same URLs.

Author: ```suggestion
You can use the [--qr-code](#--qr-code) option to display QR-codes that contain the URLs..
```

> Scan the QR-codes with the device on which you wish to test your application.

Author: Does this sentence mean we're testing an application on this particular device? (just checking)

> Scan the QR-codes with the device on which you wish to test your application.

Reviewer: Scan the QR-codes with the device you want to use to test your application.

> Scan the QR-codes with the device on which you wish to test your application.

Author: ```suggestion
Scan the QR-codes with the device you want to use to test your application.
```

### -r \<name\[:output\],\[...\]\>, --reporter \<name\[:output\],\[...\]\>

Specifies the name of a [built-in](../guides/concepts/reporters.md) or [custom reporter](../extending-testcafe/reporter-plugin/README.md) that generates test reports.

> The following command runs tests in all available browsers and generates a report in `xunit` format:

Reviewer 2: Not related to this commit:

Boris recently said that `xunit` isn't a valid **format** name, it's `JUnit XML` (check this with Boris in case I'm mistaken).

`xunit` is just our reporter's name.

Author: I believe the format itself is called `XUnit XML`. 
The tool it was created for is called JUnit, but `XUnit` is a valid format name, I think
[Source](https://google.github.io/rich-test-results/xunitxml#logo:~:text=The%20XUnit%20XML%20format,-was)

Reviewer 2: Okay, but isn't it better to just check it with Boris?

Reviewer 3: Wikipedia says it's the collective name for several unit testing frameworks.
I believe xunit is a xml subset of xunit.net and junit-like formats. In our documentation context xunit is a correct name since it has its own specs, so @DIRECTcut is right.

testcafe all tests/sample-fixture.js -r my-reporter
```

The command prompt window displays the test report generated.

To save the report to an external file, specify the file path after the report name.

```sh
testcafe all tests/sample-fixture.js -r json:report.json
```

You can use multiple reporters in a single test run. List them separated by commas.

```sh
testcafe all tests/sample-fixture.js -r spec,xunit:report.xml
```

> Note that one reporter can write to `stdout` at one time. All other reporters must output to files.

Reviewer 2: Or `at a time`?

Or better... We are not talking about writing simultaneously, we just mean "during the same test run".

`during the same run`?

Author: > During a test, only one reporter can write to `stdout`. Redirect other reporters' output to files.

> Note that one reporter can write to `stdout` at one time. All other reporters must output to files.

Author: ```suggestion
During a test, only one reporter can write to `stdout`. If you use additional reporters, redirect their output to files.
```

After the timeout passes or the `window.load` event is raised (whichever happens first), TestCafe starts the test.

> > Note that the `DOMContentLoaded` event is raised after the HTML document is loaded and parsed, while `window.load` is raised after all stylesheets, images, and subframes are loaded. That is why there is a delay between the `DOMContentLoaded` and `window.load` events fire.

Reviewer: - "while `window.load` is raised after all stylesheets"? Why do you use 'while'? Do you mean 'the `DOMContentLoaded` event is raised after the HTML document is loaded and parsed while `window.load` is raised'? But then, how can you add 'after all stylesheets, images, and subframes are loaded'? 
- "That is why there is a delay between the `DOMContentLoaded` and `window.load` events fire" Do you mean 'That is why there is a delay between **when** the `DOMContentLoaded` and `window.load` events fire'? There is also no need to add phrases like 'That is why' - This causes a delay between **when** the `DOMContentLoaded` and `window.load` events fire.

Author: This sentence is broken. Does this work:

> The `DOMContentLoaded` event is raised after the HTML document is loaded and parsed. The `window.load` is raised after all stylesheets, images, and subframes are loaded. That causes a delay between the `DOMContentLoaded` and `window.load` events.

OR

> The `DOMContentLoaded` event is raised after the HTML document is loaded and parsed. The `window.load` is raised after all stylesheets, images, and subframes are loaded. That is why there is a delay between the `DOMContentLoaded` and `window.load` events.

Reviewer: The DOMContentLoaded event is raised after the HTML document is loaded and parsed. The window.load is raised after all stylesheets, images, and subframes are loaded. This causes a delay between the DOMContentLoaded and window.load events.

> > Note that the `DOMContentLoaded` event is raised after the HTML document is loaded and parsed, while `window.load` is raised after all stylesheets, images, and subframes are loaded. That is why there is a delay between the `DOMContentLoaded` and `window.load` events fire.

Author: ```suggestion
> The `DOMContentLoaded` event is raised after the HTML document is loaded and parsed. The `window.load` event is raised after all stylesheets, images, and subframes are loaded. This causes a delay between the `DOMContentLoaded` and `window.load` events.
```

Specifies the test execution speed.

> Tests run at the maximum speed by default. You can use this option

Reviewer: - Why do you use "by default"?
- "You can use this option to slow the test down" -> You can use this option to adjust/change test speed

Author: You're right, this sentence violates guidelines.

Does this work: 

> If this parameter is not specified, tests run at the maximum speed. You can use this option to decrease test speed.

Reviewer: Ok

> Tests run at the maximum speed by default. You can use this option

Author: ```suggestion
If this parameter is not specified, tests run at the maximum speed. You can use this option to decrease test speed.
```

> to slow the test down.

Author: ```suggestion
```

testcafe chrome my-tests --speed 0.1
```

> The action's speed setting overrides the test speed specified for an individual action.

Reviewer 2: This doesn't make sense.

`The action's speed setting` and `the test speed specified for an individual action` are the same thing.

You might have misread the original sentence (or the sentence was too vague) 😄

Author: > If you specify speed for individual action, this setting overrides the global test speed.

> The action's speed setting overrides the test speed specified for an individual action.

Author: ```suggestion
If you specify speed for individual action, this setting overrides the global test speed.
```

> If you specify speed for individual action, this setting overrides the global test speed.

Reviewer: ```suggestion
If you specify the speed for an individual action, this setting overrides the global test speed.
```

Specifies custom port numbers TestCafe uses to perform testing. The number range is [0-65535].

> If you don't specify ports, TestCafe selects them within the range.

Reviewer: ```suggestion
If you do not specify ports, TestCafe automatically selects ports in the range.
```
